drv/i2c-devices/src/mwocp68.rs

```diff
@@ -350,6 +440,262 @@ impl Mwocp68 {
         Ok(val)
     }
 
+    /// Will return true if the device is present and valid -- false otherwise
+    pub fn present(&self) -> bool {
+        match Mwocp68::validate(&self.device) {
+            Ok(valid) => valid,
+            _ => false,
+        }
+    }
+
+    pub fn power_good(&self) -> Result<bool, Error> {
+        use commands::mwocp68::STATUS_WORD::*;
+
+        let status = pmbus_read!(self.device, STATUS_WORD)?;
+        Ok(status.get_power_good_status() == Some(PowerGoodStatus::PowerGood))
+    }
+
+    pub fn firmware_revision(&self) -> Result<Mwocp68FirmwareRev, Error> {
+        let mut data = [0u8; MWOCP68_REVISION_LEN];
+        let expected = MWOCP68_REVISION_FORMAT;
+
+        let len = self
+            .device
+            .read_block(CommandCode::MFR_REVISION as u8, &mut data)
+            .map_err(|code| Error::BadFirmwareRevRead { code })?;
+
+        //
+        // Per ACAN-114, we are expecting this to be of the format:
+        //
+        //    XXXX-YYYY-0000
+        //
+        // Where XXXX is the firmware revision on the primary MCU (AC input
+        // side) and YYYY is the firmware revision on the secondary MCU (DC
+        // output side).  We aren't going to be rigid about the format of
+        // either revision, but we will be rigid about the rest of the format.
```

Comment:
nitpicky: perhaps the `MWOCP_REVISION_FORMAT` constant ought to be defined near this comment? I notice that it's not referenced anywhere else, so we could just move it in here?